docs/fabric-native-components-android.md

```diff
@@ -0,0 +1,439 @@
+---
+id: fabric-native-components-android
+title: 'Fabric Native Modules: Android'
+---
+
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
+
+React Native uses Gradle to manage dependencies, auto-linking and codegen. To use this, we're going to add the `RTNCenteredText` Fabric Native Component to our `Demo` application.
+
+### 1. Link RTNCenteredText as an npm dependency:
+
+This creates symlinks between the `RTNCenteredText` component and `Demo`, so we can develop and use the component.
+
+<Tabs groupId="package-manager" queryString defaultValue={constants.defaultPackageManager} values={constants.packageManagers}>
+<TabItem value="npm">
+```bash
+cd Demo
+npm link ../RTNCenteredText
+```
+</TabItem>
+<TabItem value="yarn">
+```bash
+cd RTNCenteredText
+yarn link
+cd ../Demo
+yarn link rtn-centered-text
+```
+</TabItem>
+</Tabs>
+
+We also need to help [Metro](https://metrobundler.dev/docs/configuration/#watchfolders) understand where to find packages that are symlinked through the `node_modules`:
+
+```js title="App/metro.config.js"
+const {
+  getDefaultConfig,
+  mergeConfig,
+} = require('@react-native/metro-config');
+const path = require('path');
+
+/**
+ * Metro configuration
+ * https://reactnative.dev/docs/metro
+ *
+ * @type {import('metro-config').MetroConfig}
+ */
+const config = {
+  watchFolders: [
+    // highlight-next-line
+    path.resolve(__dirname, '../RTNCenteredText'),
+  ],
+};
+
+module.exports = mergeConfig(getDefaultConfig(__dirname), config);
+```
+
+### 2. Run Codegen through Gradle
+
+We're running this once to generate boilerplate code that'll make it easier for you to work in the IDE of your choice.
+
+```bash title="Demo/"
+cd android
+./gradlew generateCodegenArtifactsFromSchema
+```
+
+### 3. Write your Native Platform Code
+
+At the end of this, you'll have the following Android Code:
+
+```
+android
+├── build.gradle[.kts]
+└── src
+    └── main
+        └── java
+            └── com
+                └── rtncenteredtext
+                    ├── CenteredText.{kt,java]
+                    ├── CenteredTextManager.{kt,java}
+                    └── CenteredTextPackage.{kt,java}
+```
+
+Create your directories:
+
+```bash
+mkdir -p android/src/main/java/com/rtncenteredtext
+```
+
+#### Add build.gradle
+
+<Tabs groupId="android-language" queryString defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
+<TabItem value="java">
```

Comment:
I think we can have solo one of the two gradle files. Either in groovy or in kotlin. I believe that the default is kotlin, but we can remove this step easily.